app/main.py

```diff
@@ -1,3 +1,50 @@
-def shop_trip():
-    # write your code here
-    pass
+import json
+
+from app.car import Car
+from app.customer import Customer
+from app.shop import Shop
+
+
+def shop_trip() -> None:
+
+    with open("app/config.json", "r") as file:
+        config = json.load(file)
+    fuel_price = config["FUEL_PRICE"]
+
+    shops = [Shop(**shop) for shop in config["shops"]]
+    customers = [Customer(**{**customer, "car": Car(**customer["car"])})
+                 for customer in config["customers"]]
+
+    for customer in customers:
+        print(f"{customer.name} has {customer.money} dollars")
+        cheapest_shop = None
+        minimum_total_cost = float("inf")
+
+        for shop in shops:
+            trip_cost = customer.calculate_trip_cost(shop.location, fuel_price)
+            products_cost = shop.calculate_products_cost(customer.product_cart)
+            total_cost = trip_cost + products_cost
+
+            print(f"{customer.name}'s trip to the {shop.name} "
+                  f"costs {total_cost:.2f}")
+
+            if total_cost < customer.money and total_cost < minimum_total_cost:
+                cheapest_shop = shop
+                minimum_total_cost = total_cost
```

Comment:
You can create a separate function for this

Reply:
You can put all logic that makes the shop trip in a customer class and then all you do in shop_trip() function is to call this method on customer instance

Reply:
A good design here would be to keep only reading json, creating customers and calling customer methods here. This way you will structure your code better.